### ISSUE TITLE
future.Cancel causes InvalidStateError in Python 3.8.4

### ISSUE THREAD

**Issue by zkagin** [noxdafox/pebble#75]
When cancelling a ProcessFuture in Python 3.8.4, it causes an error and does not actually seem to cancel the job (though the state is Cancelled). I clarify the Python version because InvalidStateError was only recently added to concurrent.futures, which might be causing this problem.

```
Traceback (most recent call last):
  File "/home/user/downloads/miniconda/envs/main-env/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self.run()
  File "/home/user/downloads/miniconda/envs/main-env/lib/python3.8/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "/home/user/downloads/miniconda/envs/main-env/lib/python3.8/site-packages/pebble/concurrent/process.py", line 112, in _worker_handler
    future.set_exception(result)
  File "/home/user/downloads/miniconda/envs/main-env/lib/python3.8/concurrent/futures/_base.py", line 539, in set_exception
    raise InvalidStateError('{}: {!r}'.format(self._state, self))
concurrent.futures._base.InvalidStateError: CANCELLED: <ProcessFuture at 0x7fda64f61df0 state=cancelled>
```

Some sample code to trigger this:
```
from pebble import concurrent
from time import sleep

@concurrent.process
def test_func():
    sleep(20)

future = test_func()
sleep(1)
future.cancel()
```

**Comment by noxdafox** [comment]
Hello,

this is something which I observed on Travis test results myself. Unfortunately, I am struggling to reproduce such issue on my local workstations. As soon as I manage to reproduce the issue, I will provide a proper fix. Sorry for the inconvenience.

**Comment by zkagin** [comment]
Thanks @noxdafox. Is it working correctly for you on Python 3.8+? From the links below, it was only added then. I've locally patched pebble to ignore just this error, but that's probably not the best long-term solution.

https://docs.python.org/3/library/concurrent.futures.html?highlight=invalidstateerror#concurrent.futures.InvalidStateError
https://github.com/python/cpython/commit/0a28c0d12ee7201de039ced4d815f57f1f8fd48c

**Comment by noxdafox** [comment]
Tests were passing on travis for Python > 3.8 but I saw warnings similar to yours and when I tried to run these tests on my local environment I could not see them for some reason.

Your simple code snippet nevertheless, allowed me to easily reproduce the issue and fix it correctly. I do agree with you that silencing an error is seldomly a good idea on the long term.

The issue has been fixed and will make it into the next release. Thanks for reporting this!
